core/src/solvers/pipeline_planner.cpp

```diff
@@ -67,9 +67,14 @@ PipelinePlanner::PipelinePlanner() {
 	                    planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC);
 }
 
-void PipelinePlanner::init(const core::RobotModelConstPtr& robot_model) {
-	planner_ = Task::createPlanner(robot_model);
+PipelinePlanner::PipelinePlanner(const planning_pipeline::PlanningPipelinePtr& planning_pipeline) : PipelinePlanner() {
+	planner_ = planning_pipeline;
+}
 
+void PipelinePlanner::init(const core::RobotModelConstPtr& robot_model) {
+	if (!planner_) {
+		planner_ = Task::createPlanner(robot_model);
```

Comment:
In case an existing PlanningPipeline is used, it should be checked that the RobotModelPtrs are identical and an exception should be raised if not.

Reply:
Do you know how could I check if two robot_models are identical .? 

Should I add two equality operators .? 
`bool operator==(const srdf::Model& lhs, const srdf::Model& rhs)` 
`bool operator==(const urdf::ModelInterface& lhs, const urdf::ModelInterface& rhs)`

I don't think this's a good approach since ModelInterface have a lot of elements and this means adding the equality operator to each element type

Thanks!

Reply:
My suggestion is to just test for equality of the RobotModel (shared) pointers: they should represent the very same model. Do you think this is too restrictive?

Reply:
Done, Thanks!